baseplate/secrets/fetcher.py

```diff
@@ -228,6 +280,8 @@ def main():
         "vault": {
             "url": config.String,
             "role": config.String,
+            "auth_type": config.Optional(config.String, default="aws_ec2"),
```

Comment:
one of the goals of baseplate's config stuff is to make the app fail as fast as possible on configuration mistakes. to that end, getting more strict on the config parser here would allow us to ignore the possibility of configuration errors downstream (wrt your question above about error handling). i'd recommend something like:

```python
auth_types = {
    "aws": VaultClientFactory._vault_aws_ec2_auth,
    "kubernetes": VaultClientFactory._vault_kubernetes_auth,
}

...
config.Optional(config.OneOf(auth_types), default=auth_types["aws"]),
...
```

then you just pass that function along to the class and it does it from there

(:no_bell: just going with "aws" there since they've renamed the backend and it's shorter)

Reply:
It feels strange for things outside of `VaultClientFactory` to be aware of `_vault_aws_ec2_auth`. Where would be the best place to define auth types?